core/src/main/java/org/apache/carbondata/core/indexstore/row/UnsafeDataMapRow.java

```diff
@@ -308,10 +315,27 @@ private int getSizeInBytes(int ordinal, int position) {
   }
 
   private int getPosition(int ordinal) {
-    int position = 0;
+    int data = 0;
     for (int i = 0; i < ordinal; i++) {
-      position += getSizeInBytes(i, position);
+      if (dataPosition.get(i) instanceof List) {
+        data += dataPosLength((List<Object>) dataPosition.get(i));
+      } else {
+        data += (int) dataPosition.get(i);
+      }
     }
-    return position;
+    return data;
   }
+
+  private int dataPosLength(List<Object> dataPosition) {
+    int sum = 0;
+    for (Object data : dataPosition) {
+      if (data instanceof List) {
+        sum += dataPosLength((List<Object>) data);
+      } else {
+        sum += (int) data;
```

Comment:
does this sum will reach the Integer.MAX?

Reply:
no, it will not reach Integer.MAX. I have modified the existing code only. I think it will not cause any issues.